moleculemaker/clean-frontend
# feat: switch to using mmli-backend

## Problem
Legacy CLEAN backend frequently runs into problems, cant handle quoted CSV values that contain commas

## Approach
* feat: generate new API client for talking to mmli-backend
    * fix: temporary workaround to fix compile errors in generated code
* feat: adjust frontend to call new API endpoints, instead of legacy ones

## How to Test
Test with https://github.com/moleculemaker/mmli-backend/pull/45

1. Navigate to https://clean.frontend.staging.mmli1.ncsa.illinois.edu/configuration
2. Enter the following as your input string:
```
>WP_063460136
MAIPPYPDFRSAAFLRQHLRATMAFYDPVATDASGGQFHFFLDDGTVYNTHTRHLVSATRFVVTHAMLYRTTGEARYQVGMRHALEFLRTAFLDPATGGYAWLIDWQDGRATVQDTTRHCYGMAFVMLAYARAYEAGVPEARVWLAEAFDTAEQHFWQPAAGLYADEASPDWQLTSYRGQNANMHACEAMISAFRATGERRYIERAEQLAQGICQRQAALSDRTHAPAAEGWVWEHFHADWSVDWDYNRHDRSNIFRPWGYQVGHQTEWAKLLLQLDALLPADWHLPCAQRLFDTAVERGWDAEHGGLYYGMAPDGSICDDGKYHWVQAESMAAAAVLAVRTGDARYWQWYDRIWAYCWAHFVDHEHGAWFRILHRDNRNTTREKSNAGKVDYHNMGACYDVLLWALDAPGFSKESRSAALGRP
```
3. Click "Get the CLEAN Result" at the bottom-right
    * You should see a new job is created on the server: `kubectl get jobs -n staging`
    * You should see that the name of the running job matches the Job ID shown in the UI
4. Wait for the job to complete
    * You should eventually see the CSV results are fetched, parsed, and displayed on the screen
